File: keyname-hash-generator/java/src/main/java/cloud/benchflow/commons/hashing/Hashing.java
package cloud.benchflow.commons.hashing;

import java.io.UnsupportedEncodingException;
import java.math.BigInteger;
import java.security.*;

//Class for the hashing
public class Hashing {
	
	//Number of characters for the hash
	private static int numOfCharacters = 4;
	
	//Hash with MD5
	public static String hashKey(String key) throws UnsupportedEncodingException, NoSuchAlgorithmException {
		return Hashing.MD5(key);
	}
	
	//Function to hash using MD5
	public static String MD5(String key) throws UnsupportedEncodingException, NoSuchAlgorithmException {
		byte[] bytesOfMessage = key.getBytes("UTF-8");
		MessageDigest md = MessageDigest.getInstance("MD5");
		byte[] hashBytes = md.digest(bytesOfMessage);
		String hashString = new BigInteger(1,hashBytes).toString(16);
		hashString = hashString.substring(0, numOfCharacters);
		return(hashString+"/"+key);
	}
	
}

File: keyname-hash-generator/go/src/cloud/benchflow/commons/hashing/hashing.go
package hashing

import (
	"crypto/md5"
	"encoding/hex"
)

//Number of characters in the hash
const numOfCharacters int = 4

//Hash the key with MD5
func hashKey(key string) string {
	return MD5(key)
	}

//Hash the key using the MD5 crypto package
func MD5(key string) string {
	hasher := md5.New()
    hasher.Write([]byte(key))
    hashString := hex.EncodeToString(hasher.Sum(nil))
	return (hashString[:numOfCharacters]+"/"+key)
	}